### PR TITLE
Expose initial size for resizable buffer

### PR DESCRIPTION
**Pull request by philjdf**
https://github.com/G-Research/ParquetSharp/issues/50

### CODE REVIEW

File: cpp/ResizableBuffer.cpp [hunk 1]
```diff
@@ -6,11 +6,11 @@
 
 extern "C"
 {
-	PARQUETSHARP_EXPORT ExceptionInfo* ResizableBuffer_Create(std::shared_ptr<arrow::ResizableBuffer>** buffer)
+	PARQUETSHARP_EXPORT ExceptionInfo* ResizableBuffer_Create(int64_t initialSize, std::shared_ptr<arrow::ResizableBuffer>** buffer)
```
Comment (GPSnoopy) [inline cpp/ResizableBuffer.cpp]: const?

File: csharp/ExceptionInfo.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@ namespace ParquetSharp
     internal sealed class ExceptionInfo
     {
         public delegate IntPtr GetAction<TValue>(out TValue value);
+        public delegate IntPtr GetAction<TValue, TArg0>(TArg0 arg0, out TValue value);
```
Comment (GPSnoopy) [inline csharp/ExceptionInfo.cs]: Swap TArg0 and TValue to be consistent with GetFunction<in TArg0, TValue> below (also notice the in keyword)?

File: cpp/ResizableBuffer.cpp (hunk 1 above)
Comment (philjdf) [reply cpp/ResizableBuffer.cpp]: const.

File: csharp/ExceptionInfo.cs (hunk 2 above)
Comment (philjdf) [reply csharp/ExceptionInfo.cs]: Swapped.

**Pull request merged**